Lib/sqlite3/test/userfunctions.py

```diff
@@ -275,6 +276,24 @@ def CheckAnyArguments(self):
         val = cur.fetchone()[0]
         self.assertEqual(val, 2)
 
+    @unittest.skipIf(sqlite.sqlite_version_info < (3, 8, 3), "deterministic parameter not supported")
+    def CheckFuncDeterministic(self):
+        mock = unittest.mock.Mock()
```

Comment:
Why a mock is used?

Reply:
It's used to test how much times function is called. If function is non-deterministic, query can't be optimized, so function have to be called 2 times (and actually there is no need to call it at all for this query :-).
This test relies on SQLite's behavior in some degree, but I did't find out a better way to test it.

Reply:
Thank you for clarification.

Is it worth to add a test for `deterministic=False` (or for not specified)?

Reply:
Perhaps we can do something like the following instead of using mock:

```py
def foo():
    foo.calls.append(True)
foo.calls = []
```

Then we can check the call count as:

```py
self.assertEqual(len(foo.calls), 1)
```

Reply:
@serhiy-storchaka added test for it.
@berkerpeksag not sure why this could be better.

Reply:
@berkerpeksag This is how I would write this test. But I just don't use mocks much. I think using a mock here is good.